modules/aql/arrow-query-language/src/main/kotlin/arrow/aql/Count.kt

```diff
@@ -0,0 +1,27 @@
+package arrow.aql
+
+import arrow.core.*
+import arrow.data.ForListK
+import arrow.data.fix
+import arrow.data.k
+import arrow.instances.monoid
+import arrow.typeclasses.Foldable
+
+interface Count<F> {
+
+  fun foldable(): Foldable<F>
+
+  fun <A, Z> Query<F, A, Z>.count(): Query<ForListK, Long, Long> =
+    foldable().run {
+        Query(
+          select = ::identity,
+          from = listOf(from.size(Long.monoid())).k()
+        )
+    }
+
+  fun Query<ForListK, Long, Long>.value(): Long =
+    foldable().run {
```

Comment:
Is there a need to change scope to Foldable? This run is just changing scope right?